debezium-testing/debezium-testing-openshift/src/main/java/io/debezium/testing/openshift/tools/databases/db2/DB2Controller.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.testing.openshift.tools.databases.db2;
+
+import io.debezium.testing.openshift.tools.ConfigProperties;
+import io.debezium.testing.openshift.tools.WaitConditions;
+import io.debezium.testing.openshift.tools.databases.DatabaseInitListener;
+import io.debezium.testing.openshift.tools.databases.SqlDatabaseClient;
+import io.debezium.testing.openshift.tools.databases.SqlDatabaseController;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.dsl.ExecWatch;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.awaitility.Awaitility;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static io.debezium.testing.openshift.tools.ConfigProperties.DATABASE_DB2_DBZ_DBNAME;
+import static io.debezium.testing.openshift.tools.ConfigProperties.DATABASE_DB2_DBZ_PASSWORD;
+import static io.debezium.testing.openshift.tools.ConfigProperties.DATABASE_DB2_DBZ_USERNAME;
+import static io.debezium.testing.openshift.tools.ConfigProperties.DATABASE_POSTGRESQL_DBZ_DBNAME;
+import static org.awaitility.Awaitility.await;
+
+/**
+ *
+ * @author Jakub Cechacek
+ */
+public class DB2Controller extends SqlDatabaseController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DB2Controller.class);
+    private static final String READINESS_SQL_SELECT = "SELECT 1 FROM DB2INST1.CUSTOMERS;";
+
+    public DB2Controller(Deployment deployment, List<Service> services, String dbType, OpenShiftClient ocp) {
+        super(deployment, services, dbType, ocp);
+    }
+
+
+    public void initialize() {
+        LOGGER.info("Waiting until DB2 instance is ready");
```

Comment:
Hey @SeanRooooney, we could need some helps on Db2 :) Specifically,@jcechace here is looking for a way to ensure the database has started (as a container) in integration tests before running any test. I.e. some readiness probe of sort. We do specific selects e.g. for Postgres in order to achieve that, and are looking for any Db2 equivalent. Thanks for any advise.

Reply:
The issue at hand is not exactly about detecting the DB startup. It's more about why the statement execution fails.

Reply:
it looks like not in a readiness state. You can connect, but not all of the preparing work is done. About IBM Db2 on Openshift will be probably found here https://hub.docker.com/r/ibmcom/db2 .